docs/stats/lombscargle.rst

```diff
@@ -415,6 +418,191 @@ For more information on the statistical properties of these normalizations,
+signal in the data, we will observe a peak this high or higher approximately
+0.4% of the time, which gives a strong indication that a periodic signal is
+present in the data.
+
+.. Note::
+  One must be careful in interpreting this probability: it is a measurement
+  conditioned on the assumption of the null hypothesis of no signal; in symbols,
+  one might write :math:`P({\rm data} \mid {\rm noise-only})`.
+
+  It is tempting to interpret this quantity incorrectly; in particular, you
+  might wish to say a statement like "there is an 0.4% chance that this data
+  is noise only", but this is *not* a correct statement; in symbols, this
+  statement describes the quantity :math:`P({\rm noise-only} \mid {\rm data})`,
+  and in general :math:`P(A\mid B) \ne P(B\mid A)`.
+
+  See [11]_ for a more detailed discussion of such caveats.
+
+We might also wish to compute the required peak height to attain any given
+false alarm probability, which can be done with the
+:func:`~astropy.stats.LombScargle.false_alarm_level` method:
+
+.. doctest-requires:: scipy
+
+  >>> probabilities = [0.1, 0.05, 0.01]
+  >>> ls.false_alarm_level(probabilities)  # doctest: +FLOAT_CMP
+  array([ 0.25446627,  0.27436154,  0.31716182])
+
+This tells us that to attain a 10% false alarm probability requires the highest
+periodogram peak to be approximately 0.25; 5% requires 0.27, and 1% requires
+0.32.
+
+False Alarm Approximations
+--------------------------
+
+Although the false alarm probability at any particular frequency is analytically
+computable, there is no closed-form analytic expression for the more relevant
+quantity of the false alarm level of the *highest* peak in a particular
+periodogram.
+This must be either determined through bootstrap simulations, or approximated
+by various means.
+
+AstroPy provides four options for approximating the false alarm probability,
+which can be chosen using the ``method`` keyword:
+
+- ``method="baluev"`` (the default)
+  implements the approximation proposed by Baluev 2008 [8]_,
+  which employs extreme value statistics to compute an upper-bound of the false
+  alarm probability for the alias-free case. Experiments show that the bound is
+  also useful even for highly-aliased observing patterns.
+
+  .. doctest-requires:: scipy
+
+    >>> ls.false_alarm_probability(power.max(), method='baluev')  # doctest: +FLOAT_CMP
+    0.0043217866919174324
+
+- ``method="bootstrap"`` implements a bootstrap simulation: effectively it
+  computes many Lomb-Scargle periodograms on simulated data at the same
+  observation times. The bootstrap approach can very accurately determine
+  the false alarm probability, but is very computationally expensive:
+  to estimate the level corresponding to a false alarm probability
+  :math:`P_{false}`, it requires on order :math:`n_{boot} \approx 10/P_{false}`
+  individual periodograms to be computed for the dataset.
+
+  .. doctest-requires:: scipy
```

Comment:
and here.  Note that this doctest is skipped below, so this `.. doctest-requires` is really unnecessary.